scout/ecm_prep.py

```diff
@@ -13088,6 +13076,42 @@ def tsv_cost_carb_yrmap(tsv_data, aeo_years):
     return tsv_yr_map
 
 
+def downselect_packages(existing_pkgs: list[dict], pkg_subset: list) -> list:
+    downselected_pkgs = [pkg for pkg in existing_pkgs if pkg["name"] in pkg_subset]
+
+    return downselected_pkgs
+
+
+def filter_invalid_packages(packages: list[dict],
+                            ecms: list,
+                            pkgs_filtered: bool) -> list[dict]:
+    """Identify and filter packages whose ECMs are not all present in the individual ECM set
+
+    Args:
+        packages (list[dict]): List of packages imported from package_ecms.json
+        ecms (list): List of ECM definitions file names
+        pkgs_filtered (bool): Indicate whether the packages have been filtered via the
+            ecm_packages argument
+
+    Returns:
+        filtered_packages (list[dict]): Packages list with invalid packages filtered out
+    """
+
+    invalid_pkgs = [pkg["name"] for pkg in packages if not
+                    set(pkg["contributing_ECMs"]).issubset(set(ecms))]
+    if invalid_pkgs:
+        package_opt_txt = ""
+        if pkgs_filtered:
+            package_opt_txt = "specified with the ecm_packages argument "
+        msg = (f"WARNING: Packages in package_ecms.json {package_opt_txt}have contributing ECMs"
```

Comment:
Is it possible to improve the appearance of the printed list of skipped packages? It would be nice if the list of packages printed with each package name on a new line (and possibly if the preceding warning text also appeared on its own line).

Reply:
Corrected in https://github.com/trynthink/scout/pull/372/commits/b362cacf4e12614ee5771cfe1ebaebf40a40dd1b